server/container_list.go

```diff
@@ -31,35 +31,33 @@ func (s *Server) ListContainers(ctx context.Context, req *pb.ListContainersReque
 	logrus.Debugf("ListContainersRequest %+v", req)
 	var ctrs []*pb.Container
 	filter := req.Filter
-	ctrList := s.state.containers.List()
+	ctrList, _ := s.state.GetAllContainers()
 
 	// Filter using container id and pod id first.
 	if filter != nil {
 		if filter.Id != "" {
-			id, err := s.ctrIDIndex.Get(filter.Id)
+			c, err := s.state.LookupContainerByID(filter.Id)
```

Comment:
The behaviour here changed. It used to be the case that if filter.Id did not exist, we returned an empty list, which seems right. Now we return an error. We should probably add a special error for container not found, and return the empty list in that case, and the error otherwise (e.g. for the internal error we have in LookupContainerByid() atm)